sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -283,10 +294,15 @@ class SessionCatalog(
    * explicitly specified.
    */
   def isTemporaryTable(name: TableIdentifier): Boolean = {
-    !name.database.isDefined && tempTables.contains(formatTableName(name.table))
+    name.database.isEmpty && tempTables.contains(formatTableName(name.table))
   }
 
   /**
+   * Return whether View is supported
+   */
+  def isViewSupported: Boolean = false
```

Comment:
Since we move `DROP TABLE` to sql/core, users are allowed to `DROP VIEW` in `SQLContext`. Thus, the parser is unable to detect it. Both are sharing the same function.